### ISSUE TITLE
Behavior of Thread::new() on passed instances

### ISSUE THREAD

**Issue by pbosetti** [mattn/mruby-thread#39]
The original implementation (see also #38) is designed so that `Thread.new(a)` makes a full copy of `a` in the newly created thread (unless `mrb_type(a) == MRB_TT_DATA`). 
As a consequence, changes to `a` in any given thread are not visible to the other threads. This way, passing information between threads becomes a hard task.

Since I am still not sure about the reasons behind this choice, I have made an experimental branch in my own fork ([pbosetti/mruby-thread](/pbosetti/mruby-thread/tree/experimental)) where the C macro `MRB_THREAD_COPY_VALUES` can be set to enable the original behavior (make copies) or leave undefined to switch to a "share the same instances" behavior.

The script `examples/data.rb` can be used to compare the two cases: when `MRB_THREAD_COPY_VALUES` is set, the changes to globals `$data` and `$ary` made in the subthread are not visible into the main thread. Conversely, if `MRB_THREAD_COPY_VALUES`, any change to instances passed to `Thread::new()` are visible to other threads.

The question is: can I make a PR for this?

Thanks,
Paolo

**Comment by pbosetti** [comment]
Hello @mattn any update on this?

**Comment by mattn** [comment]
Sorry, I'll look into it in later.

**Comment by pbosetti** [comment]
@mattn no problem, didn't want to put pressure, just wanted to be suer you got the notification. More than often it happened to me 😉

**Comment by mattn** [comment]
> The question is: can I make a PR for this?
 Yes, please do it. I hope that mruby-thread can handle instances created by another thread.
